network/p2p/dht_test.go

```diff
@@ -202,67 +184,13 @@ func (suite *DHTTestSuite) TestPubSubWithDHTDiscovery() {
 					missing = append(missing, n.id)
 				}
 			}
-			assert.Fail(suite.T(), " messages not received by nodes: "+strings.Join(missing.Strings(), ","))
+			assert.Fail(t, " messages not received by nodes: "+strings.Join(missing.Strings(), ","))
 			break
 		}
 	}
 
 	// Step 6: unsubscribes all nodes from the topic
 	for _, n := range nodes {
-		assert.NoError(suite.T(), n.UnSubscribe(topic))
-	}
-}
-
-// CreateNode creates the given number of libp2pnodes
-// if dhtServer is true, the DHTServer is used as for Discovery else DHTClient
-func (suite *DHTTestSuite) CreateNodes(count int, dhtServer bool) (nodes []*Node) {
```

Comment:
high repetition logic with already existing `nodeFixture` hence combined to be more DRY (Don't Repeat Yourself)